src/common/settings.ts

```diff
@@ -332,6 +333,56 @@ function getPreferredGlobalSetting<T>(
   return newSettings?.defaultValue;
 }
 
+/**
+ * Check if the user have configured `notebook.codeActionsOnSave` with non-notebook prefixed code actions.
+ */
+export function checkNotebookCodeActionsOnSave(serverId: string) {
+  getWorkspaceFolders().forEach((workspace) => {
+    let codeActionsOnSave: string[] = (() => {
+      const value = getConfiguration("notebook", workspace.uri).get<string[] | object>(
+        "codeActionsOnSave",
+        [],
+      );
+      if (typeof value === "object") {
+        return Object.keys(value);
+      }
+      return value;
+    })();
+
+    const genericCodeActions = codeActionsOnSave.filter(
+      (action) => action === "source.organizeImports" || action === "source.fixAll",
+    );
+
+    const ruffCodeActions = codeActionsOnSave.filter(
+      (action) =>
+        action === `source.organizeImports.${serverId}` || action === `source.fixAll.${serverId}`,
+    );
+
+    if (genericCodeActions.length > 0) {
+      // This is at info level because other extensions might be using these code actions but we still want to inform the user.
+      logger.info(
+        `The following code actions in 'notebook.codeActionsOnSave' could lead to unexpected behavior: ${JSON.stringify(
+          genericCodeActions,
+        )}. Consider using ${JSON.stringify(
+          genericCodeActions.map((action) => `notebook.${action}`),
+        )} instead.`,
+      );
```

Comment:
It seems you also added a similar check on the LSP side. Do we only include the warning here because this is a VS code specific fix or is this something we could handle on the LSP side?

Should we include a link to some documentation explaining the why? E.g., we could add a FAQ entry or a section on the website that explains why using the notebook specific actions is preferred.

Reply:
> It seems you also added a similar check on the LSP side. Do we only include the warning here because this is a VS code specific fix or is this something we could handle on the LSP side?

Yes, this is VS Code specific fix as we're looking into `notebook.codeActionsOnSave` which specific to VS Code and can provide suggestion to change it.

> Should we include a link to some documentation explaining why? For example, we could add an FAQ entry or a section on the website explaining why using notebook-specific actions is preferred. It might also be could to link to the issue, but that's something we could do in the FAQ section where we have a bit more space than here.

Thanks, that's a great suggestion. I'll add that first.